### PR TITLE
looking-glass-client/kvmfr: B5.0.1 -> B6

### PR DESCRIPTION
**Pull request by j-brn**
###### Description of changes

- updated looking-glass-client and kvmfr to B6
- added parameters to optionally disable X11, Wayland, Pulseaudio or Pipewire support
- removed backported patches for kvmfr that are already included in the new release

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by wyndon — COMMENTED** [review]
Works great in my Windows VM.

**Review by alexbakker — APPROVED** [review]

File: pkgs/applications/virtualization/looking-glass-client/default.nix [hunk 1]
```diff
@@ -16,40 +48,32 @@ let
...
+    ++ lib.optionals waylandSupport [ libxkbcommon wayland wayland-protocols ]
+    ++ lib.optionals pipewireSupport [ pipewire libsamplerate ]
+    ++ lib.optionals pulseSupport [ pulseaudio libsamplerate ];
+
+  cmakeFlags =
+    [ "-DOPTIMIZE_FOR_NATIVE=OFF" ]
+    ++ lib.optionals (!xorgSupport) [ "-DENABLE_X11=no" ]
+    ++ lib.optionals (!waylandSupport) [ "-DENABLE_WAYLAND=no" ]
+    ++ lib.optionals (!pulseSupport) [ "-DENABLE_PULSEAUDIO=no" ]
+    ++ lib.optionals (!pipewireSupport) [ "-DENABLE_PIPEWIRE=no" ];
```
Comment (babbaj) [inline pkgs/applications/virtualization/looking-glass-client/default.nix:75]: `lib.optional` would be slightly simpler

**Review by babbaj — APPROVED** [review]

File: pkgs/applications/virtualization/looking-glass-client/default.nix (hunk 1 above)
Comment (j-brn) [reply pkgs/applications/virtualization/looking-glass-client/default.nix]: Good catch! Changed in 0a0a60db08f5ddd7202e5e02ef169f1c58ee31ea

**Review by alexbakker — APPROVED** [review]

**Comment by j-brn** [comment]
Is there anything that's blocking this right now?

**Comment by LightningRhino** [comment]
Is there a problem with the new package?

**Comment by j-brn** [comment]
> Is there a problem with the new package?

No, the new version works fine, I'm already using it. Just waiting for someone to approve and merge this

File: pkgs/applications/virtualization/looking-glass-client/default.nix [hunk 2]
```diff
@@ -16,40 +48,32 @@ let
...
-  ];
+  buildInputs =
+    [ libGL libX11 freefont_ttf spice-protocol expat libbfd nettle fontconfig libffi ]
+    ++ lib.optionals xorgSupport [ libxkbcommon libXi libXScrnSaver libXinerama libXcursor libXpresent libXext libXrandr ]
+    ++ lib.optionals waylandSupport [ libxkbcommon wayland wayland-protocols ]
+    ++ lib.optionals pipewireSupport [ pipewire libsamplerate ]
+    ++ lib.optionals pulseSupport [ pulseaudio libsamplerate ];
+
+  cmakeFlags =
+    [ "-DOPTIMIZE_FOR_NATIVE=OFF" ]
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/looking-glass-client/default.nix:71]:
```suggestion
  buildInputs = [ libGL libX11 freefont_ttf spice-protocol expat libbfd nettle fontconfig libffi ]
    ++ lib.optionals xorgSupport [ libxkbcommon libXi libXScrnSaver libXinerama libXcursor libXpresent libXext libXrandr ]
    ++ lib.optionals waylandSupport [ libxkbcommon wayland wayland-protocols ]
    ++ lib.optionals pipewireSupport [ pipewire libsamplerate ]
    ++ lib.optionals pulseSupport [ pulseaudio libsamplerate ];

  cmakeFlags = [ "-DOPTIMIZE_FOR_NATIVE=OFF" ]
```
nit

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The 1st and 3rd commit belong together. The 4th could also be combined.

File: pkgs/os-specific/linux/kvmfr/default.nix [hunk 3]
```diff
@@ -9,19 +9,6 @@ stdenv.mkDerivation rec {
   hardeningDisable = [ "pic" "format" ];
   nativeBuildInputs = kernel.moduleBuildDependencies;
 
-  patches = lib.optional (kernel.kernelAtLeast "5.16") (fetchpatch {
```
[the remark this answers is not in the document]
Comment (j-brn) [reply pkgs/os-specific/linux/kvmfr/default.nix:12]: done

**Comment by j-brn** [comment]
> The 1st and 3rd commit belong together. The 4th could also be combined.

cleaned up the history

**Pull request merged**